### DISCUSSION TITLE
Please add other options for transcription providers other than openai.

### DISCUSSION THREAD

**Discussion by Abdalnablse10** [WAppAI/assistant#104] [Ideas]
Openai transcription is great but sadly the free plan is limited to a 5$ credit and it would be great to have other free options.

**Comment by Luisotee** [comment]
Will do, hopefully in the upcoming weeks.

**Comment by Luisotee** [comment]
Done in #110

**Comment by Abdalnablse10** [comment]
Hey thank you the addition of whisper ai python version.

But please add the c++ version, it uses way less memory and is way faster.

The following test was done on an amlogic s905w cpu linux with 2GB of ram.

1 min and 50 sec with initializing "initializing took 20 sec", tiny model, 1GB ram, python, command used: whisper jfk.wav --model tiny

32.29175 sec initializing was instant, tiny model, 204 MB ram, c++, command used: ./main -m models/ggml-tiny.bin -f samples/jfk.wav

25.39772 sec initializing was instant, quantized tiny model, 160 MB ram, c++, ./main -m models/ggml-tiny-q5_0.bin -f samples/jfk.wav

The file used for testing was samples/jfk.wav in https://github.com/ggerganov/whisper.cpp

The whisper.cpp is built from source on the user side but the build time takes just less than 5 minutes on a low end device "the models are provided besides the quantized ones but converting the normal models to quantized takes about one minute"

**Reply by Luisotee** [reply]
Looks cool, I will try to do it.

**Reply by Abdalnablse10** [reply]
btw i don't know if this can help but i tried to add it myself "i don't know how to code :-P"
this is what i came up with in a day and failed "only the conversion using ffmpeg part worked... i think XD"
[audio-transcription-local-modified.txt](https://github.com/veigamann/sydney-whatsapp-chatbot/files/12039058/audio-transcription-local-modified.txt)
[audio-transcription-modified.txt](https://github.com/veigamann/sydney-whatsapp-chatbot/files/12039061/audio-transcription-modified.txt)
[env.example.modified.txt](https://github.com/veigamann/sydney-whatsapp-chatbot/files/12039062/env.example.modified.txt)

**Comment by Abdalnablse10** [comment]
Hello, I'd like to add that there is https://github.com/ariym/whisper-node
Basically it's whisper.cpp for nodejs.

**Reply by Luisotee** [reply]
Has been a while but just to leave a closure here. In the new refactor of the project in the branch pnpm-refactor we are using the c++ version of whisper in the project
